src/Compression/Compression.cs

```diff
@@ -666,23 +700,31 @@ public async Task CopyFromSourceToDestination ()
 			{
 				_deflateStream.AsyncOperationStarting ();
 				try {
+					_deflateStream.EnsureDecompressionMode ();
 					// Flush any existing data in the inflater to the destination stream.
 					while (true) {
-						int bytesRead = _deflateStream._inflater.Inflate (_arrayPoolBuffer, 0, _arrayPoolBuffer.Length);
+						// TODO Inflate throws exception if _arrayPoolBuffer is null, do we want to
+						// create a new _arrayPoolBuffer if null or throw an exception here?
```

Comment:
@mandel-macaque similar things happen down here as well with the `_arrayPoolBuffer`.  I am calling exceptions here if null, but we could also initialize the buffer, or just pass it in the method calls and let them throw the exception?